Footsteps sounds for bare feet and other mobs

nero1024 opened the pull request:
Bare feet and other mobs will now make the correct footstep sounds and added new samples to go with them. Added some helpers to is_helpers.dm, mostly for (future) footstep support but maybe someone else can make use of them.

:cl: Nero1024
soundadd: Bare feet will now make the correct footstep sounds.
soundadd: Other mobs will make the correct footstep sounds.
/:cl:

As others indicated in the very first footsteps PR other users wanted footstep sounds for barefeet. I also noticed that do_footstep is in a bunch of other mobs, but Ian and Xenomorphs don't wear shoes so I fixed that too.

Demos:
Barefeet in action: https://my.mixtape.moe/lrtohp.webm
Xenomrphs clacking around: https://my.mixtape.moe/edzqpc.webm


TODO:
- [x] Fix: Make dead mobs not play footsteep sounds.
- [x] Tweak: Adjust footstep volume for xenos? They're quiet, but perhaps too quiet IMO.

nero1024 committed:
Merge pull request #2 from tgstation/master

Rebase to Current

nero1024 committed:
Merge pull request #3 from tgstation/master

Update master

nero1024 committed:
Merge pull request #4 from tgstation/master

update master

nero1024 committed:
support for clawed and heavy footstep types

nero1024 committed:
more footstep types and is_helpers

nero1024 committed:
more footstep sounds

nero1024 committed:
make slimes do footstep

nero1024 committed:
more is_helper and footsteps upates

nero1024 committed:
better footsteps for mobs

Qustinnus commented:
code seems snowflakey

nero1024 commented:
I'll admit that `ishumansimple()` is an absolute mess. Any tips on how It can be tidied up and made presentable?

ShizCalev commented on code/__DEFINES/is_helpers.dm:
```diff
@@ -116,22 +116,45 @@ GLOBAL_LIST_INIT(turfs_without_ground, typecacheof(list(
+#define isgiantspider(A) (istype(A, /mob/living/simple_animal/hostile/poison/giant_spider))
+
+//Human(oid) simple mobs (it's ugly, barely readable, stupidly long, and can't into newlines)
+#define ishumansimple(A) (istype(A, /mob/living/simple_animal/hostile/cat_butcherer) || istype(A, /mob/living/simple_animal/hostile/faithless) || istype(A, /mob/living/simple_animal/hostile/nanotrasen) || istype(A, /mob/living/simple_animal/hostile/pirate) || istype(A, /mob/living/simple_animal/hostile/russian) || istype(A, /mob/living/simple_animal/hostile/syndicate) || istype(A, /mob/living/sim ...
```
atleast make it a typecache or something
https://github.com/tgstation/tgstation/blob/151c4e9b183bf233fa9755ecd0d0bb4d7896e208/code/__DEFINES/is_helpers.dm#L16-L23

ShizCalev commented on code/datums/components/footstep.dm:
```diff
@@ -27,13 +29,76 @@
+		return
+	
+	//for xenomorphs, dogs, and other clawed mobs
+	if(isalienadult(LM) || isdog(LM) || isfox(LM) || isbear(LM) || ismegaarachnid(LM))
```
typecache

Bawhoppen commented:
I like adding new footstep sounds, but I'm not so sure about the ones for xenos. While I'm not sure how they *should* sound, doesn't seem right to me.

nicbn commented:
Have an associative list including sounds for barefoot, shoes, claws etc. as a var for turfs. Use defines as keys.
Have a var for mobs and shoes that equals to the define of the sound of such mob or shoe.

When the mob steps, check for shoes. If none or if the var equals zero, use the mob define. Else, use the shoes define.
Then, access the turf list using such define as key.

No istypes, just oop.

Qustinnus commented:
why not just make the sound lists defines as we do with things such as ambience

epochayur commented:
ignoring the fact that xeno footsteps are a thinly-veiled excuse to include step sounds for lizards, isn't the entire point of xenos that they're genetically perfect silent killers? it doesn't make sense to me that you could hear them coming

Poojawa commented:
Xenos have very noisy breathing problems, also not spamming *deathgasp is a terrible Xeno

nero1024 committed:
tydy up is_helpers.dm and footsteps.dm

nero1024 committed:
tydy up footstep.dm

nero1024 committed:
Delete robot1.ogg (it sounded dumb anyways)

nero1024 committed:
this doesn't belong here

nero1024 committed:
nor this

nero1024 committed:
put updated files in the right place this time

nicbn commented on code/__DEFINES/is_helpers.dm:
```diff
@@ -116,22 +116,96 @@ GLOBAL_LIST_INIT(turfs_without_ground, typecacheof(list(
 
 #define iscat(A) (istype(A, /mob/living/simple_animal/pet/cat))
 
+#define isdog(A) (istype(A, /mob/living/simple_animal/pet/dog))
```
Remove the unused defines you added

nicbn commented on code/__DEFINES/is_helpers.dm:
```diff
@@ -116,22 +116,96 @@ GLOBAL_LIST_INIT(turfs_without_ground, typecacheof(list(
 
+#define isgiantspider(A) (istype(A, /mob/living/simple_animal/hostile/poison/giant_spider))
+
+GLOBAL_LIST_INIT(simplehuman, typecacheof(list(
```
You should use a var in these types instead of making a list.

nero1024 committed:
remove unsed defines in is_helpers.dm

Militaires commented:
footsteps for xenos blows their cover early on and ruins the earlygame stealth you need to establish a colony, do u even xenos bro

nero1024 committed:
added simple_animal alien to helpers

nero1024 committed:
made xeno footsteps quieter

nero1024 commented:
I've tried quieting xeno footsteps down, It seems like you have to be right next to them to hear them now. 
Here's a webm:
https://my.mixtape.moe/vwgqlw.webm

nero1024 committed:
add farm animals to footstep helpers

GuyonBroadway commented:
Personally I think the xenos sounds are a little too "slappy" for lack of a better term.

Needs a harder "clack" if you can do such a thing.

nero1024 commented:
Those are the clack-iest claw sounds that I can source.

nero1024 commented:
`if(!T.footstep || LM.buckled || !CHECK_MULTIPLE_BITFIELDS(LM.mobility_flags, MOBILITY_STAND | MOBILITY_MOVE) || LM.throwing || LM.movement_type & (VENTCRAWLING | FLYING))`, specifically mobility_flags, is supposed to check if the mob cannot stand or move correct? How come I'm still hearing footstep sounds from mobs when they're crawling, dead, or lying and being dragged?

nero1024 committed:
incap'd/dead mobs wont make footstep sounds

Also tweaked xeno footstep sound distance a bit. Footstep sounds still happen while crawling though.

nero1024 commented:
I've decided to keep the footstep sounds while crawling, and have the barefeet sounds play as well when crawling (cuz you're using your hands to crawl).

nero1024 committed:
barefeet sounds will play when crawling cuz hands

Militaires commented:
ok what if you're wearing gloves

nero1024 commented:
It would sound about the same.
